provider/pkg/gen/schema.go

```diff
@@ -69,7 +69,7 @@ func PulumiSchema(swagger map[string]interface{}) pschema.PackageSpec {
 					TypeSpec:    pschema.TypeSpec{Type: "string"},
 				},
 				"deleteUnreachable": {
-					Description: "If present and set to true, the provider will delete resources associated with an unreachable Kubernetes cluster from Pulumi state",
+					Description: "If present and set to true, the provider will delete resources associated with an unreachable Kubernetes cluster from Pulumi state. It can also be sourced from the following environment variable: `PULUMI_K8S_DELETE_UNREACHABLE`",
```

Comment:
If you add `.DefaultInfo.Environment` in the `.Provider` section of the schema, I believe that will also update the API docs, and add env var handling to the SDKs.

https://github.com/pulumi/pulumi-kubernetes/pull/2312/files#diff-4b4c0cf25563429565383180f98332b1dd806ba3fc7b7cd8473ba51435199667R136-R139